test/Interop/Cxx/stdlib/use-std-iterator.swift

```diff
@@ -0,0 +1,46 @@
+// RUN: %target-run-simple-swift(-I %S/Inputs -Xfrontend -enable-cxx-interop)
+//
+// REQUIRES: executable_test
+//
+// Enable this everywhere once we have a solution for modularizing libstdc++: rdar://87654514
+// REQUIRES: OS=macosx
+
+import StdlibUnittest
+import StdIterator
+import std.iterator
+import StdVector
+import std.vector
+
+var StdIteratorTestSuite = TestSuite("StdIterator")
+
+StdIteratorTestSuite.test("init") {
+    var vector = Vector()
+    var v1 = Iterator()
+    var _1: CInt = 1
+    vector.push_back(&_1)
+    v1 = begin(vector)
+    expectEqual(v1[0], 1)
+}
```

Comment:
```suggestion
StdIteratorTestSuite.test("init") {
    var vector = Vector()
    vector.push_back(1)
    let it = vector.begin()
    expectEqual(it[0], 1)
}
```

Reply:
There is an interesting problem that's exposed by the above program around copies. I think what you'll have to do is either copy vector into a let constant or use beginMutating to prevent a copy of self before begin is invoked. This is something we're going to have to look into how to properly solve.. it's a huge issue.

Reply:
ohhh i see, that makes sense